src/Aspire.Hosting/Health/ResourceHealthCheckService.cs

```diff
@@ -76,11 +85,20 @@ await eventing.PublishAsync(
                 if (!resourceReadyEventFired && report.Status == HealthStatus.Healthy)
                 {
                     resourceReadyEventFired = true;
-                    var resourceReadyEvent = new ResourceReadyEvent(resource, services);
-                    await eventing.PublishAsync(
-                        resourceReadyEvent,
-                        EventDispatchBehavior.NonBlockingSequential,
-                        cancellationToken).ConfigureAwait(false);
+
+                    _ = Task.Run(async () =>
```

Comment:
Is this the same as above? What do you think of adding a local method to reduce duplication?

Reply:
Will do